internal/stoppropaganda/websites.go

```diff
@@ -275,6 +288,27 @@ func doSingleRequest(ws *Website, req *fasthttp.Request, resp *fasthttp.Response
 
 	resp.ShouldDiscardBody = true
 
+	if *flagRandomQuery {
+		// clean up query string and cookies from previous request
+		req.URI().SetQueryString("")
```

Comment:
You don't need to clean anything here. :)

Copy of request pointer (*fasthttp.Request) is being done here: https://github.com/erkexzcx/stoppropaganda/blob/main/internal/stoppropaganda/websites.go#L239

You will always have a fresh, original and unmodified copy of request in each function call, so you can remove those 2 lines.

Reply:
Ah okay! This was needed when I was modifying `ws` directly but not any more